docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -67,15 +67,38 @@ By default, the name of the decorated function, `my_asset`, is used as the asset
 
 ### Assets with dependencies
 
-Software-defined assets can depend on other software-defined assets. In this section, we'll show you how to define:
+Software-defined assets can depend on other software-defined assets. In Dagster, you can either define dependencies that just specify materialization order, or you can define dependencies that also load the contents of the upstream asset. You can also define dependencies on external assets.
+
+In this section, we'll show you how to define:
 
 - [Basic asset dependencies](#defining-basic-dependencies)
-- [Explicit asset dependencies](#defining-explicit-dependencies)
+- [Basic data-loading asset dependencies](#defining-explicit-dependencies)
+- [Explicit data-loading asset dependencies](#defining-explicit-dependencies)
 - [External asset dependencies](#defining-external-asset-dependencies)
 
 #### Defining basic dependencies
 
-The easiest way to define an asset dependency is to include an upstream asset name as an argument to the decorated function.
+You can define a dependency between two assets by passing the upstream asset to the `deps` parameter in the downstream asset's `@asset` decorator.
+
+In this example, `upstream_asset` creates a new table (`sugary_cereals`) by selecting records from the `cereals` table. Then `downstream_asset` creates a new table (`shopping_list`) by selecting records from `sugary_cereals`
+
+```python file=/concepts/assets/non_argument_deps.py startafter=start_marker endbefore=end_marker
+from dagster import asset
+
+
+@asset
+def upstream_asset() -> None:
+    execute_query("CREATE TABLE sugary_cereals AS SELECT * FROM cereals")
+
+
+@asset(deps=[upstream_asset])
+def downstream_asset() -> None:
+    execute_query("CREATE TABLE shopping_list AS SELECT * FROM sugary_cereals")
+```
+
+#### Defining basic data-loading dependencies
+
+The easiest way to define an asset dependency that will load the contents of an upstream asset is to include the upstream asset name as an argument to the decorated function.
```

Comment:
Yeah this elaboration is good, one qq thought

> it's expected that the code you include inside your @asset--decorated function will load the data from the upstream asset.  

that's not quite true though right? because you might just be running SQL queries on the table created by the upstream asset, or launching some external process that then uses the upstream asset

Reply:
Yeah I guess I'm being a little loose here. When you write "select ...", that does cause data to be loaded, it just happens in the database instead of the process executing your Python code. With Spark, it's even more of a gray area: you actually write `spark.read` inside Python, but the read is usually executed remotely.  There's probably some way of being more precise, it's not coming to me immediately though.

Reply:
> it's expected that the code you include inside your @asset-decorated function will access the data from the upstream asset

I like this phrasing

> For example, you may run a SELECT query against the upstream asset, or a spark.read that executes remotely. If you need direct access to the contents of the upstream asset in memory, then you are also responsible for loading the contents of the asset within the @asset-decorated function.

"or a spark.read that executes remotely" makes it sound like you wouldn't do it for the case where a spark.read executes locally. I would maybe just present a couple values equally: "For example, you may run a SQL statement that selects from the upstream table, load the upstream CSV file into a Pandas DataFrame".

Reply:
what if we left it more generic and said "for example, you may want to fetch the data into memory, or you may want to execute a command that works with the data in an external process"